LabTools/CoreWidgets/CommandWidget.py

```diff
@@ -0,0 +1,861 @@
+                #funct = self.process_params(funct)[0] if you want function to be dynamic
+            except (ValueError, IndexError, TypeError): #no params
+                funct = command
+                params = None
+                pass
+            #split params
+            try:
+                params = self.process_params(params) #replace important stuff
+            except (ValueError, IndexError, TypeError): #1 param
+                params = [params]
+                pass
+            except:
+                params = []
+                pass
+
+            #now actually call the command
+            try:
+                resp = getattr(self.sanitized_list[current_device][2], funct)(*params)
+                self.update_console("Function returned with: "+self.print_to_string(resp))
+            except AttributeError: #errors likely caused by run syntax
+                self.update_console(self.print_to_string("Failed to run command: Attribute Error. Likely invalid function name or command syntax"))
+                self.update_console(self.print_to_string("Use 'methods' to obtain list of callable functions"))
+                self.update_console(self.print_to_string("Use 'run function paramater1 parameter2 etc' to execute function"))
+                if self.INFO:
+                    self.update_console(self.print_to_string(sys.exc_info()))
+                pass
+            except: #any other errors
+                self.update_console(self.print_to_string("Failed to run command: ",sys.exc_info()[0]))
+                if self.INFO:
+                    self.update_console(self.print_to_string(sys.exc_info()))
+                pass
+        elif action.lower() == "plot":
+            # generally going to be the same as methods, but will initialize matplotlib plot
+            try:
+                funct, params = command.split(' ', 1)
+            except (ValueError, IndexError, TypeError): #no params
+                funct = command
+                params = None
+                pass
+            #split params
+            try:
+                params = self.process_params(params) #replace important stuff
+            except (ValueError, IndexError, TypeError): #1 param
+                params = [params]
+                pass
+            except:
+                params = []
+                pass
+
+            #now actually call the command
+            try:
+                resp = getattr(self.sanitized_list[current_device][2], funct)(*params)
+            except AttributeError: #errors likely caused by run syntax
+                self.update_console(self.print_to_string("Failed to run command: Attribute Error. Likely invalid function name or command syntax"))
+                self.update_console(self.print_to_string("Use 'methods' to obtain list of callable functions"))
+                self.update_console(self.print_to_string("Use 'run function paramater1 parameter2 etc' to execute function"))
```

Comment:
copy-paste of the line 266-268, could not put that in a function?

Reply:
It would probably be a good idea to have a seperate method/object that deals with handling errors like this, so that the actual text can be easily changed (for updates, translations, etc etc)